ddtrace/context.py

```diff
@@ -241,3 +240,39 @@ def get(self):
             self._locals.context = ctx
 
         return ctx
+
+
+class _ContextVarContext(object):
+    """
+    _ContextVarContext can be used in place of the ThreadLocalContext for Python
+    3.7 and above to manage different ``Context`` objects for each thread and
+    async task.
+    """
+
+    def __init__(self):
+        self.token = None
+
+    def _has_active_context(self):
+        ctx = DATADOG_CONTEXT.get()
+        return ctx is not None
+
+    def set(self, ctx):
+        self.token = DATADOG_CONTEXT.set(ctx)
+
+    def get(self):
+        ctx = DATADOG_CONTEXT.get()
+        if not ctx:
+            ctx = Context()
+            self.set(ctx)
+
+        return ctx
+
+
+try:
+    from contextvars import ContextVar
+
+    DATADOG_CONTEXT = ContextVar('datadog_contextvar', default=None)
+    ContextManager = _ContextVarContext
+except ImportError:
+    DATADOG_CONTEXT = None
```

Comment:
I almost wonder if we shouldn't just have:

```python
try:
    import contextvars
   
    class ContextVarsContext(object):
        pass
except ImportError:
    ContextVarsContext = None
```

Then in places we can have:

```python
from ddtrace.context import ThreadLocalContext, ContextVarsContext

if ContextVarsContext:
    ctx = ContextVarsContext()
else:
    ctx = ThreadLocalContext()
```

I don't love this idea, just brainstorming some other possibilities.

Reply:
My thinking so far has been that everywhere but the context module, none of the code should need to know we are using `contextvars` or not, since they'll operate just like thread-local storage (hence replacing `ThreadLocalContext` references with `ContextManager`).

Though we do have a need in the asyncio code to check which is enabled at time of patching to add the contextvars-specific wrapper. So I'm open to a cleaner way to expose the different underlying context manager.